pkg/util/fast_int_set.go

```diff
@@ -17,10 +17,11 @@ package util
 import (
 	"bytes"
 	"fmt"
+	"sort"
 )
 
-// FastIntSet keeps track of a set of integers. It is very fast when the values
-// are small. It is not thread-safe.
+// FastIntSet keeps track of a set of integers. It does not perform any
+// allocations when the values are small. It is not thread-safe.
 type FastIntSet struct {
 	smallVals uint64
 	largeVals map[uint32]struct{}
```

Comment:
So given the range of values that's likely to occur, from experience this is much more efficient:
- `largeVals []int64`
- set with `largeVals[i/64] |= 1 << (i%64)`
- check with `largeVals[i/64] & ( 1 << (i%64) )`
- you can also skip 8 bits at a time in the same way you do for smallVals